packages/web-runtime/src/components/SidebarNav/SidebarNav.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div id="web-nav-sidebar" :class="sidebarClasses">
+    <oc-button
+      variation="inverse"
+      appearance="raw"
+      :class="toggleSidebarButtonClass"
+      class="toggle-sidebar-button oc-py-s"
+      :aria-label="$gettext('Toggle sidebar')"
+      @click="toggleSidebarButtonClick"
+    >
+      <oc-icon size="large" :name="toggleSidebarButtonIcon" />
+    </oc-button>
+    <nav
+      class="oc-sidebar-nav oc-my-l oc-px-xs"
+      :aria-label="$gettext('Sidebar top navigation menu')"
+    >
+      <oc-list>
+        <sidebar-nav-item
+          v-for="(link, index) in navItems"
+          :id="`nav-item-${index}`"
+          :key="link.route.path"
+          :target="link.route.path"
+          :active="link.active"
+          :icon="link.icon || link.iconMaterial"
+          :name="link.name"
+          :collapsed="navigation.closed"
+        />
+      </oc-list>
+    </nav>
+    <!-- @slot bottom content of the sidebar -->
+    <slot name="bottom" />
+  </div>
+</template>
+
+<script>
+import { mapState, mapActions } from 'vuex'
+import SidebarNavItem from './SidebarNavItem.vue'
+
+export default {
+  components: {
+    SidebarNavItem
+  },
+  props: {
+    navItems: {
+      type: Array,
+      required: true
+    }
+  },
+  computed: {
+    ...mapState(['sidebar', 'navigation']),
+
+    sidebarClasses() {
+      if (this.navigation.closed) {
```

Comment:
can be simplified by having `uk-visible@l` as default and add additional classes conditional